### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by amaanq**
Mostly changes I consider to be more correct/appropriate

### CODE REVIEW

**Comment by clason** [comment]
That's not a lot to go on... Can you _please_ explain the changes and _why_ they are more appropriate?

(Remember that this parser is a crucial bundled part of the Neovim infrastructure now, so I have to be extra careful.)

**Comment by amaanq** [comment]
Yeah sure, sorry

- Rename list to alternation, it's the more correct term
 -Add string_content to strings, it's useful for injecting/extracting the content
 - Make definition a supertype, it's still hidden from the parse tree but is queryable and makes more sense for it

**Comment by clason** [comment]
Indent queries have to be adapted too. And please test the effect on the builtin query linter and -- especially -- on nvim-treesitter's format-queries.

File: grammar.js [hunk 1]
```diff
@@ -20,41 +20,37 @@ module.exports = grammar({
     /\s+/,
   ],
 
+  supertypes: $ => [
+    $.definition,
+  ],
```
Comment (lucario387) [inline grammar.js:25]: I don't really understand much about the parsing side, but doesn't this mean supertypes are now definitions, so something like `(a/(b (c)))` are now possible?

**Comment by clason** [comment]
Ok, I think I have to put on the brakes here; I _really_ don't like "Updates: things I like better". Can you separate each of these changes into individual PRs so we can review, discuss and (for non-breaking improvements) merge separately? (And yes, that means generating the parser for each change. I strongly believe that _if_ you track `parser.c`, it should be consistent with the repo state -- otherwise bisecting becomes impossible.

File: grammar.js (hunk 1 above)
Comment (amaanq) [reply grammar.js:25]: Yeah, where `a` is `definition`, you can query for any of the subtypes of definition using the `/` syntax

Though it'd be like `(a/b (c))` iirc

**Pull request closed without merging**